src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -236,6 +357,10 @@ class CoreWorkerDirectTaskSubmitter {
     // Tasks that are queued for execution. We keep an individual queue per
     // scheduling class to ensure fairness.
     std::deque<TaskSpecification> task_queue = std::deque<TaskSpecification>();
+    // Keep a queue of tasks that have not been used yet to request a new worker from the
+    // raylet.
+    std::deque<TaskSpecification> tasks_to_request_workers =
```

Comment:
Instead of keeping this extra queue, could you try overwriting the task ID with a random ID? Actually, even now the task ID does not necessarily match the ID of tasks assigned to that worker. I think the change should be simple enough that we can do it in this PR.

Reply:
Just changed this in the commit: https://github.com/ray-project/ray/pull/13570/commits/2a8e15d11d77f396c9f822d51d6313ee9a718592